scipy/stats/tests/test_stats.py

```diff
@@ -3793,6 +3794,28 @@ def test_ttest_rel():
     assert_raises(ValueError, stats.ttest_rel, x.reshape((8, 3)),
                   x.reshape((2, 3, 4)))
 
+    # Convert from two-sided p-values to one sided using T result data.
+    def convert(t, p, alt):
+        if (t < 0 and alt == "less") or (t > 0 and alt == "greater"):
+            return p / 2
+        return 1 - (p / 2)
+    converter = np.vectorize(convert)
+
+    rvs1_2D[:, 20:30] = np.nan
+    rvs2_2D[:, 15:25] = np.nan
+
+    tr, pr = stats.ttest_rel(rvs1_2D, rvs2_2D, 0, nan_policy='omit')
+
+    t, p = stats.ttest_rel(rvs1_2D, rvs2_2D, 0, nan_policy='omit',
+                           alternative='less')
+    assert_allclose(t, tr, atol=1e-12)
+    assert_allclose(p, converter(tr, pr, 'less'), atol=1e-12)
```

Comment:
Why `atol` here? If you're trying to make the tolerance more strict, use `rtol`. 
If you need it to be looser, why? If it can't pass `rtol`, we should probably not have a test with such small p-values (because with `atol` you can lose the distinction between the two-sided and one-sided p-values).

Reply:
> Why `atol` here? If you're trying to make the tolerance more strict, use `rtol`.

Thanks for the suggestion. Done.



> If you need it to be looser, why? If it can't pass `rtol`, we should probably not have a test with such small p-values (because with `atol` you can lose the distinction between the two-sided and one-sided p-values).

Oh, didn't know that atol loosens the test. I will keep this in mind from now on :)

Reply:
If the reference p-value were exactly zero, then it would be impossible to get the desired relative tolerance unless the actual result were exactly zero. Adding an absolute tolerance, then, would effectively loosen the tolerance. That's all I meant.